### PR TITLE
Reusable blocks: Add to Block Manager

### LINKED ISSUE #13390: Find a better place for the "Manage All Reusable Blocks" link
In the top bar "more" menu, "Manage All Reusable Blocks" is a link and triggers navigation to `wp-admin/edit.php?post_type=wp_block`:

screenshot 2019-01-20 at 19 07 36

However, it's placed within an ARIA `menu` and has a role of `menuitem`. This way, it's announced by assistive technologies as a menu item. There's nothing to inform users it's a link and that activating it will trigger navigation to a new page. In short, the current interaction is completely unexpected.

I'd say that also visually there's nothing that communicates to users this is a link. The navigation to a new page may be completely unexpected for non screen reader users too.

Using an `<a>` element is correct, but this link can't be placed within a role=menu. This is a good example where design needs to take semantics and expected interaction into consideration. The whole menu is made of menu items for _actions_ and that's the convention communicated to users. A link feels out of place. 

Worth noting that in the inserter, this link is actually communicated as a link because the native semantics is preserved:

screenshot 2019-01-20 at 19 22 17

ARIA roles and attributes communicate semantics and also expected interaction. It would be great to enforce via code that components using role=menu can't contain links.

Regarding the specific case of the "Manage All Reusable Blocks" link, it should be removed from the more menu entirely.

I seem to recall there was some previous conversation about the discoverability of this link. As it's a link that points to a default WordPress admin page that displays a specific post type, one option could be considering to place it in the WordPress admin menu on the left.

### PR DESCRIPTION
**Pull request by gziolo**
## Description
Closes #13390.

WIP

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

**Comment by mapk** [comment]
The "Manage All Reusable Blocks" link has been removed from the Options popover.

Screen Shot 2020-01-03 at 4 57 24 PM

The Reusable blocks category is now added to the Block Manager along with the "Edit blocks" link.

Screen Shot 2020-01-03 at 4 58 04 PM

File: packages/edit-post/src/components/manage-blocks-modal/style.scss [hunk 1]
```diff
@@ -75,6 +75,10 @@
 		font-size: 0.9rem;
 		font-weight: 600;
 	}
+
+	> a {
+		float: right; //Aligning link to the right of category heading.
```
Comment (gziolo) [inline packages/edit-post/src/components/manage-blocks-modal/style.scss]: It looks like using `float` changes only the visual appearance but not the tabbing order:
tab-order-block-manager

We need to find a different approach here.

Comment (mapk) [reply packages/edit-post/src/components/manage-blocks-modal/style.scss]: Yeah, I was wondering about that. `float:right` works easier this way, but tab order is off now. I'll fix this.

**Comment by mapk** [comment]
I've added `flexbox` to the category title to improve the tab order.

tab-order 2020-01-07 12_43_28

File: packages/edit-post/src/components/manage-blocks-modal/category.js [hunk 2]
```diff
@@ -66,14 +67,16 @@ function BlockManagerCategory( {
 			aria-labelledby={ titleId }
 			className="edit-post-manage-blocks-modal__category"
 		>
-			<div className="edit-post-manage-blocks-modal__category-title">
+			<div className="edit-post-manage-blocks-modal__catąegory-title">
```
Comment (gziolo) [inline packages/edit-post/src/components/manage-blocks-modal/category.js]: @mapk - I know why styles are broken, I accidentally typed polish character :)

**Comment by gziolo** [comment]
@mapk, I'm nearly there. I still need to apply filtering for Reusable Blocks in the inserter. Block Manager seems to work as intended. I didn't test it very extensively, I will resume tomorrow.

Screen Shot 2020-02-05 at 17 12 49
Screen Shot 2020-02-05 at 17 12 59

**Comment by gziolo** [comment]
I must admit that I didn't anticipate the amount of refactoring necessary to apply this simple change. First, we had to ensure that reusable blocks are exposed in the Block Manager. I wrongly assumed that they are there. Now, it’s tricky to filter out hidden reusable blocks from the inserter, because the existing logic depends on the list of allowed blocks that can be exposed from the server and then merged with those hidden in the Block Manager by users.

It’s also very likely that we will want to make it possible to hide newly introduced block variations from the inserter as well. Ideally, we tackle reusable blocks in a way that makes it easy to follow-up with block variations. I prefer to not rush with this feature and include it first in the regular flow where it lands in the Gutenberg plugin first and is properly tested.

**Comment by mapk** [comment]
How's this one going, @gziolo ? Is there anything I can help with?

**Comment by gziolo** [comment]
> How's this one going, @gziolo ? Is there anything I can help with?

I need to revisit it, there is some challenge in resolving how settings are propagated between internal packages. I hope to get back to it later this month.

**Comment by mapk** [comment]
@gziolo any updates on this one?

**Comment by gziolo** [comment]
@mapk I was busy with other tasks, it's still on my todo list 👍

**Comment by gziolo** [comment]
I'm starting extended leave soon (3 months) and I won't be able to finish this task until I'm back. If anyone wants to take over this task, feel free to do so 🥇

**Comment by afercia** [comment]
Seems I can't test this PR: there's no "Reusable blocks" section in the modal for me. I guess this PR is a bit behind.

**Comment by gziolo** [comment]
@afercia I'm afraid you are right. Let's close it and start fresh.

**Pull request closed without merging**